### PR TITLE
Colw/transaction service

### PR DESCRIPTION
**Pull request by faboweb**
**Description:**

~Requires updated Lunie-API that fetches accountNum and sequence.~ Updated.

Set `VUE_APP_ENABLE_TX_API` to `true` to enable.

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: src/ActionModal/utils/ActionManager.js [hunk 1]
```diff
@@ -12,17 +12,14 @@ const txFetchOptions = {
...
-
 async function transactionAPIRequest(payload) {
   // console.log(config)
   const options = {
     ...txFetchOptions,
     body: JSON.stringify({ payload })
   }
 
-  return fetch(`${config.graphqlHost}/transaction`, options).then(r => r.json())
+  return fetch(`${config.graphqlHost}/transaction/estimate`, options).then(r => r.json())
```
Comment (faboweb) [inline src/ActionModal/utils/ActionManager.js]: we have axios as a dependency. would you rather prefer using fetch?

Comment (colw) [reply src/ActionModal/utils/ActionManager.js]: I would personally. It's supported in all modern browsers.

The reason I used it here was just convenience though to get something running.

File: src/ActionModal/utils/MessageConstructor.js [hunk 2]
```diff
@@ -46,3 +61,167 @@ export const getMultiMessage = async (context, messages) => {
     }
   }
 }
+
+// Bank
+export function MsgSend(
```
Comment (faboweb) [inline src/ActionModal/utils/MessageConstructor.js]: Better export these functions from the library instead?

Comment (colw) [reply src/ActionModal/utils/MessageConstructor.js]: Agreed, but they are not currently exported from the library. I can do this at a later time.

**Comment by faboweb** [comment]
@colw maybe add some coverage to ActionManager?

File: src/ActionModal/components/ActionModal.vue [hunk 3]
```diff
@@ -618,7 +619,16 @@ export default {
       const { type, memo, ...properties } = this.transactionData
       await this.actionManager.setMessage(type, properties)
       try {
-        this.gasEstimate = await this.actionManager.simulate(memo)
+        if (!config.enableTxAPI) {
+          this.gasEstimate = await this.actionManager.simulate(memo)
+        } else {
+          this.gasEstimate = await this.actionManager.simulateTxAPI(
+            this.createContext(),
```
Comment (jbibla) [inline src/ActionModal/components/ActionModal.vue]: what does this do @colw?

**Comment by jbibla** [comment]
action modal is so large!

File: src/ActionModal/components/ActionModal.vue (hunk 3 above)
Comment (colw) [reply src/ActionModal/components/ActionModal.vue]: The old functionality is still there, and you can opt-in to the transaction service by setting an environment variable when building the front end.

Comment (faboweb) [reply src/ActionModal/components/ActionModal.vue]: he meant the context maybe. that is like a bunch of variables you need for the tx to be build

**Comment by colw** [comment]
> action modal is so large!

What do you mean?

File: src/ActionModal/components/ActionModal.vue (hunk 3 above)
Comment (jbibla) [reply src/ActionModal/components/ActionModal.vue]: yep, that's what i meant. but i don't see the function anywhere.

**Comment by jbibla** [comment]
> What do you mean? Or rather, in which way?

so many lines!

File: src/ActionModal/components/ActionModal.vue (hunk 3 above)
Comment (colw) [reply src/ActionModal/components/ActionModal.vue]: It's around, line ~500. Not part of this PR though.

**Pull request merged**